// @unkey/api/dist/commonjs/sdk/ratelimit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Ratelimit = void 0;
const ratelimitDeleteOverride_js_1 = require("../funcs/ratelimitDeleteOverride.js");
const ratelimitGetOverride_js_1 = require("../funcs/ratelimitGetOverride.js");
const ratelimitLimit_js_1 = require("../funcs/ratelimitLimit.js");
const ratelimitListOverrides_js_1 = require("../funcs/ratelimitListOverrides.js");
const ratelimitSetOverride_js_1 = require("../funcs/ratelimitSetOverride.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class Ratelimit extends sdks_js_1.ClientSDK {
    async limit(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, ratelimitLimit_js_1.ratelimitLimit)(this, request, options));
    }
    async setOverride(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, ratelimitSetOverride_js_1.ratelimitSetOverride)(this, request, options));
    }
    async getOverride(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, ratelimitGetOverride_js_1.ratelimitGetOverride)(this, request, options));
    }
    async listOverrides(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, ratelimitListOverrides_js_1.ratelimitListOverrides)(this, request, options));
    }
    async deleteOverride(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, ratelimitDeleteOverride_js_1.ratelimitDeleteOverride)(this, request, options));
    }
}
exports.Ratelimit = Ratelimit;
