// @unkey/api/dist/commonjs/mcp-server/server.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMCPServer = createMCPServer;
const mcp_js_1 = require("@modelcontextprotocol/sdk/server/mcp.js");
const core_js_1 = require("../core.js");
const prompts_js_1 = require("./prompts.js");
const resources_js_1 = require("./resources.js");
const tools_js_1 = require("./tools.js");
const apisCreateApi_js_1 = require("./tools/apisCreateApi.js");
const identitiesCreateIdentity_js_1 = require("./tools/identitiesCreateIdentity.js");
const livenessLiveness_js_1 = require("./tools/livenessLiveness.js");
const ratelimitDeleteOverride_js_1 = require("./tools/ratelimitDeleteOverride.js");
const ratelimitGetOverride_js_1 = require("./tools/ratelimitGetOverride.js");
const ratelimitLimit_js_1 = require("./tools/ratelimitLimit.js");
const ratelimitListOverrides_js_1 = require("./tools/ratelimitListOverrides.js");
const ratelimitSetOverride_js_1 = require("./tools/ratelimitSetOverride.js");
function createMCPServer(deps) {
    const server = new mcp_js_1.McpServer({
        name: "Unkey",
        version: "2.0.0-alpha.7",
    });
    const client = new core_js_1.UnkeyCore({
        rootKey: deps.rootKey,
        serverURL: deps.serverURL,
        serverIdx: deps.serverIdx,
    });
    const scopes = new Set(deps.scopes);
    const allowedTools = deps.allowedTools && new Set(deps.allowedTools);
    const tool = (0, tools_js_1.createRegisterTool)(deps.logger, server, client, scopes, allowedTools);
    const resource = (0, resources_js_1.createRegisterResource)(deps.logger, server, client, scopes);
    const resourceTemplate = (0, resources_js_1.createRegisterResourceTemplate)(deps.logger, server, client, scopes);
    const prompt = (0, prompts_js_1.createRegisterPrompt)(deps.logger, server, client, scopes);
    const register = { tool, resource, resourceTemplate, prompt };
    void register; // suppress unused warnings
    tool(ratelimitLimit_js_1.tool$ratelimitLimit);
    tool(ratelimitSetOverride_js_1.tool$ratelimitSetOverride);
    tool(ratelimitGetOverride_js_1.tool$ratelimitGetOverride);
    tool(ratelimitListOverrides_js_1.tool$ratelimitListOverrides);
    tool(ratelimitDeleteOverride_js_1.tool$ratelimitDeleteOverride);
    tool(identitiesCreateIdentity_js_1.tool$identitiesCreateIdentity);
    tool(apisCreateApi_js_1.tool$apisCreateApi);
    tool(livenessLiveness_js_1.tool$livenessLiveness);
    return server;
}
