// @unkey/api/dist/esm/funcs/livenessLiveness.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as components from "../models/components/index.js";
import * as errors from "../models/errors/index.js";
import { APIPromise } from "../types/async.js";
/**
 * Liveness check
 *
 * @remarks
 * This endpoint checks if the service is alive.
 */
export function livenessLiveness(client, options) {
    return new APIPromise($do(client, options));
}
async function $do(client, options) {
    const path = pathToFunc("/v2/liveness")();
    const headers = new Headers(compactMap({
        Accept: "application/json",
    }));
    const secConfig = await extractSecurity(client._options.rootKey);
    const securityInput = secConfig == null ? {} : { rootKey: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "liveness",
        oAuth2Scopes: [],
        resolvedSecurity: requestSecurity,
        securitySource: client._options.rootKey,
        retryConfig: options?.retries
            || client._options.retryConfig
            || {
                strategy: "backoff",
                backoff: {
                    initialInterval: 50,
                    maxInterval: 1000,
                    exponent: 1.5,
                    maxElapsedTime: 10000,
                },
                retryConnectionErrors: true,
            }
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["5XX"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["412", "4XX", "500", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result] = await M.match(M.json(200, components.V2LivenessResponseBody$inboundSchema), M.jsonErr(412, errors.PreconditionFailedErrorResponse$inboundSchema), M.jsonErr(500, errors.InternalServerErrorResponse$inboundSchema), M.fail("4XX"), M.fail("5XX"))(response, { extraFields: responseFields });
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
