// @unkey/api/dist/commonjs/models/components/badrequesterrordetails.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BadRequestErrorDetails$ = exports.BadRequestErrorDetails$outboundSchema = exports.BadRequestErrorDetails$inboundSchema = void 0;
exports.badRequestErrorDetailsToJSON = badRequestErrorDetailsToJSON;
exports.badRequestErrorDetailsFromJSON = badRequestErrorDetailsFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const validationerror_js_1 = require("./validationerror.js");
/** @internal */
exports.BadRequestErrorDetails$inboundSchema = z.object({
    detail: z.string(),
    instance: z.string().optional(),
    status: z.number().int(),
    title: z.string(),
    type: z.string(),
    errors: z.array(validationerror_js_1.ValidationError$inboundSchema),
});
/** @internal */
exports.BadRequestErrorDetails$outboundSchema = z.object({
    detail: z.string(),
    instance: z.string().optional(),
    status: z.number().int(),
    title: z.string(),
    type: z.string(),
    errors: z.array(validationerror_js_1.ValidationError$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BadRequestErrorDetails$;
(function (BadRequestErrorDetails$) {
    /** @deprecated use `BadRequestErrorDetails$inboundSchema` instead. */
    BadRequestErrorDetails$.inboundSchema = exports.BadRequestErrorDetails$inboundSchema;
    /** @deprecated use `BadRequestErrorDetails$outboundSchema` instead. */
    BadRequestErrorDetails$.outboundSchema = exports.BadRequestErrorDetails$outboundSchema;
})(BadRequestErrorDetails$ || (exports.BadRequestErrorDetails$ = BadRequestErrorDetails$ = {}));
function badRequestErrorDetailsToJSON(badRequestErrorDetails) {
    return JSON.stringify(exports.BadRequestErrorDetails$outboundSchema.parse(badRequestErrorDetails));
}
function badRequestErrorDetailsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BadRequestErrorDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BadRequestErrorDetails' from JSON`);
}
