// @unkey/api/dist/commonjs/models/components/ratelimitlimitresponsedata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RatelimitLimitResponseData$ = exports.RatelimitLimitResponseData$outboundSchema = exports.RatelimitLimitResponseData$inboundSchema = void 0;
exports.ratelimitLimitResponseDataToJSON = ratelimitLimitResponseDataToJSON;
exports.ratelimitLimitResponseDataFromJSON = ratelimitLimitResponseDataFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.RatelimitLimitResponseData$inboundSchema = z.object({
    limit: z.number().int(),
    remaining: z.number().int(),
    reset: z.number().int(),
    success: z.boolean(),
    overrideId: z.string().optional(),
});
/** @internal */
exports.RatelimitLimitResponseData$outboundSchema = z.object({
    limit: z.number().int(),
    remaining: z.number().int(),
    reset: z.number().int(),
    success: z.boolean(),
    overrideId: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RatelimitLimitResponseData$;
(function (RatelimitLimitResponseData$) {
    /** @deprecated use `RatelimitLimitResponseData$inboundSchema` instead. */
    RatelimitLimitResponseData$.inboundSchema = exports.RatelimitLimitResponseData$inboundSchema;
    /** @deprecated use `RatelimitLimitResponseData$outboundSchema` instead. */
    RatelimitLimitResponseData$.outboundSchema = exports.RatelimitLimitResponseData$outboundSchema;
})(RatelimitLimitResponseData$ || (exports.RatelimitLimitResponseData$ = RatelimitLimitResponseData$ = {}));
function ratelimitLimitResponseDataToJSON(ratelimitLimitResponseData) {
    return JSON.stringify(exports.RatelimitLimitResponseData$outboundSchema.parse(ratelimitLimitResponseData));
}
function ratelimitLimitResponseDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RatelimitLimitResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RatelimitLimitResponseData' from JSON`);
}
