// @unkey/api/dist/commonjs/models/components/v2identitiescreateidentityrequestbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2IdentitiesCreateIdentityRequestBody$ = exports.V2IdentitiesCreateIdentityRequestBody$outboundSchema = exports.V2IdentitiesCreateIdentityRequestBody$inboundSchema = exports.V2IdentitiesCreateIdentityRequestBodyMeta$ = exports.V2IdentitiesCreateIdentityRequestBodyMeta$outboundSchema = exports.V2IdentitiesCreateIdentityRequestBodyMeta$inboundSchema = void 0;
exports.v2IdentitiesCreateIdentityRequestBodyMetaToJSON = v2IdentitiesCreateIdentityRequestBodyMetaToJSON;
exports.v2IdentitiesCreateIdentityRequestBodyMetaFromJSON = v2IdentitiesCreateIdentityRequestBodyMetaFromJSON;
exports.v2IdentitiesCreateIdentityRequestBodyToJSON = v2IdentitiesCreateIdentityRequestBodyToJSON;
exports.v2IdentitiesCreateIdentityRequestBodyFromJSON = v2IdentitiesCreateIdentityRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const v2ratelimit_js_1 = require("./v2ratelimit.js");
/** @internal */
exports.V2IdentitiesCreateIdentityRequestBodyMeta$inboundSchema = z.object({});
/** @internal */
exports.V2IdentitiesCreateIdentityRequestBodyMeta$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2IdentitiesCreateIdentityRequestBodyMeta$;
(function (V2IdentitiesCreateIdentityRequestBodyMeta$) {
    /** @deprecated use `V2IdentitiesCreateIdentityRequestBodyMeta$inboundSchema` instead. */
    V2IdentitiesCreateIdentityRequestBodyMeta$.inboundSchema = exports.V2IdentitiesCreateIdentityRequestBodyMeta$inboundSchema;
    /** @deprecated use `V2IdentitiesCreateIdentityRequestBodyMeta$outboundSchema` instead. */
    V2IdentitiesCreateIdentityRequestBodyMeta$.outboundSchema = exports.V2IdentitiesCreateIdentityRequestBodyMeta$outboundSchema;
})(V2IdentitiesCreateIdentityRequestBodyMeta$ || (exports.V2IdentitiesCreateIdentityRequestBodyMeta$ = V2IdentitiesCreateIdentityRequestBodyMeta$ = {}));
function v2IdentitiesCreateIdentityRequestBodyMetaToJSON(v2IdentitiesCreateIdentityRequestBodyMeta) {
    return JSON.stringify(exports.V2IdentitiesCreateIdentityRequestBodyMeta$outboundSchema.parse(v2IdentitiesCreateIdentityRequestBodyMeta));
}
function v2IdentitiesCreateIdentityRequestBodyMetaFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2IdentitiesCreateIdentityRequestBodyMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2IdentitiesCreateIdentityRequestBodyMeta' from JSON`);
}
/** @internal */
exports.V2IdentitiesCreateIdentityRequestBody$inboundSchema = z.object({
    externalId: z.string(),
    meta: z.lazy(() => exports.V2IdentitiesCreateIdentityRequestBodyMeta$inboundSchema)
        .optional(),
    ratelimits: z.array(v2ratelimit_js_1.V2Ratelimit$inboundSchema).optional(),
});
/** @internal */
exports.V2IdentitiesCreateIdentityRequestBody$outboundSchema = z.object({
    externalId: z.string(),
    meta: z.lazy(() => exports.V2IdentitiesCreateIdentityRequestBodyMeta$outboundSchema)
        .optional(),
    ratelimits: z.array(v2ratelimit_js_1.V2Ratelimit$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2IdentitiesCreateIdentityRequestBody$;
(function (V2IdentitiesCreateIdentityRequestBody$) {
    /** @deprecated use `V2IdentitiesCreateIdentityRequestBody$inboundSchema` instead. */
    V2IdentitiesCreateIdentityRequestBody$.inboundSchema = exports.V2IdentitiesCreateIdentityRequestBody$inboundSchema;
    /** @deprecated use `V2IdentitiesCreateIdentityRequestBody$outboundSchema` instead. */
    V2IdentitiesCreateIdentityRequestBody$.outboundSchema = exports.V2IdentitiesCreateIdentityRequestBody$outboundSchema;
})(V2IdentitiesCreateIdentityRequestBody$ || (exports.V2IdentitiesCreateIdentityRequestBody$ = V2IdentitiesCreateIdentityRequestBody$ = {}));
function v2IdentitiesCreateIdentityRequestBodyToJSON(v2IdentitiesCreateIdentityRequestBody) {
    return JSON.stringify(exports.V2IdentitiesCreateIdentityRequestBody$outboundSchema.parse(v2IdentitiesCreateIdentityRequestBody));
}
function v2IdentitiesCreateIdentityRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2IdentitiesCreateIdentityRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2IdentitiesCreateIdentityRequestBody' from JSON`);
}
