// @unkey/api/dist/commonjs/models/components/v2livenessresponsebody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2LivenessResponseBody$ = exports.V2LivenessResponseBody$outboundSchema = exports.V2LivenessResponseBody$inboundSchema = void 0;
exports.v2LivenessResponseBodyToJSON = v2LivenessResponseBodyToJSON;
exports.v2LivenessResponseBodyFromJSON = v2LivenessResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const livenessresponsedata_js_1 = require("./livenessresponsedata.js");
const meta_js_1 = require("./meta.js");
/** @internal */
exports.V2LivenessResponseBody$inboundSchema = z.object({
    meta: meta_js_1.Meta$inboundSchema,
    data: livenessresponsedata_js_1.LivenessResponseData$inboundSchema,
});
/** @internal */
exports.V2LivenessResponseBody$outboundSchema = z.object({
    meta: meta_js_1.Meta$outboundSchema,
    data: livenessresponsedata_js_1.LivenessResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2LivenessResponseBody$;
(function (V2LivenessResponseBody$) {
    /** @deprecated use `V2LivenessResponseBody$inboundSchema` instead. */
    V2LivenessResponseBody$.inboundSchema = exports.V2LivenessResponseBody$inboundSchema;
    /** @deprecated use `V2LivenessResponseBody$outboundSchema` instead. */
    V2LivenessResponseBody$.outboundSchema = exports.V2LivenessResponseBody$outboundSchema;
})(V2LivenessResponseBody$ || (exports.V2LivenessResponseBody$ = V2LivenessResponseBody$ = {}));
function v2LivenessResponseBodyToJSON(v2LivenessResponseBody) {
    return JSON.stringify(exports.V2LivenessResponseBody$outboundSchema.parse(v2LivenessResponseBody));
}
function v2LivenessResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2LivenessResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2LivenessResponseBody' from JSON`);
}
