// @unkey/api/dist/commonjs/models/components/v2ratelimitlimitrequestbody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2RatelimitLimitRequestBody$ = exports.V2RatelimitLimitRequestBody$outboundSchema = exports.V2RatelimitLimitRequestBody$inboundSchema = void 0;
exports.v2RatelimitLimitRequestBodyToJSON = v2RatelimitLimitRequestBodyToJSON;
exports.v2RatelimitLimitRequestBodyFromJSON = v2RatelimitLimitRequestBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.V2RatelimitLimitRequestBody$inboundSchema = z.object({
    namespace: z.string(),
    cost: z.number().int().default(1),
    duration: z.number().int(),
    identifier: z.string(),
    limit: z.number().int(),
});
/** @internal */
exports.V2RatelimitLimitRequestBody$outboundSchema = z.object({
    namespace: z.string(),
    cost: z.number().int().default(1),
    duration: z.number().int(),
    identifier: z.string(),
    limit: z.number().int(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2RatelimitLimitRequestBody$;
(function (V2RatelimitLimitRequestBody$) {
    /** @deprecated use `V2RatelimitLimitRequestBody$inboundSchema` instead. */
    V2RatelimitLimitRequestBody$.inboundSchema = exports.V2RatelimitLimitRequestBody$inboundSchema;
    /** @deprecated use `V2RatelimitLimitRequestBody$outboundSchema` instead. */
    V2RatelimitLimitRequestBody$.outboundSchema = exports.V2RatelimitLimitRequestBody$outboundSchema;
})(V2RatelimitLimitRequestBody$ || (exports.V2RatelimitLimitRequestBody$ = V2RatelimitLimitRequestBody$ = {}));
function v2RatelimitLimitRequestBodyToJSON(v2RatelimitLimitRequestBody) {
    return JSON.stringify(exports.V2RatelimitLimitRequestBody$outboundSchema.parse(v2RatelimitLimitRequestBody));
}
function v2RatelimitLimitRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2RatelimitLimitRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitLimitRequestBody' from JSON`);
}
