// @unkey/api/dist/commonjs/models/components/v2ratelimitlistoverridesresponsebody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2RatelimitListOverridesResponseBody$ = exports.V2RatelimitListOverridesResponseBody$outboundSchema = exports.V2RatelimitListOverridesResponseBody$inboundSchema = void 0;
exports.v2RatelimitListOverridesResponseBodyToJSON = v2RatelimitListOverridesResponseBodyToJSON;
exports.v2RatelimitListOverridesResponseBodyFromJSON = v2RatelimitListOverridesResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const meta_js_1 = require("./meta.js");
const pagination_js_1 = require("./pagination.js");
const ratelimitoverride_js_1 = require("./ratelimitoverride.js");
/** @internal */
exports.V2RatelimitListOverridesResponseBody$inboundSchema = z.object({
    meta: meta_js_1.Meta$inboundSchema,
    data: z.array(ratelimitoverride_js_1.RatelimitOverride$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema.optional(),
});
/** @internal */
exports.V2RatelimitListOverridesResponseBody$outboundSchema = z.object({
    meta: meta_js_1.Meta$outboundSchema,
    data: z.array(ratelimitoverride_js_1.RatelimitOverride$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2RatelimitListOverridesResponseBody$;
(function (V2RatelimitListOverridesResponseBody$) {
    /** @deprecated use `V2RatelimitListOverridesResponseBody$inboundSchema` instead. */
    V2RatelimitListOverridesResponseBody$.inboundSchema = exports.V2RatelimitListOverridesResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitListOverridesResponseBody$outboundSchema` instead. */
    V2RatelimitListOverridesResponseBody$.outboundSchema = exports.V2RatelimitListOverridesResponseBody$outboundSchema;
})(V2RatelimitListOverridesResponseBody$ || (exports.V2RatelimitListOverridesResponseBody$ = V2RatelimitListOverridesResponseBody$ = {}));
function v2RatelimitListOverridesResponseBodyToJSON(v2RatelimitListOverridesResponseBody) {
    return JSON.stringify(exports.V2RatelimitListOverridesResponseBody$outboundSchema.parse(v2RatelimitListOverridesResponseBody));
}
function v2RatelimitListOverridesResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2RatelimitListOverridesResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitListOverridesResponseBody' from JSON`);
}
