// @unkey/api/dist/commonjs/models/components/v2ratelimitsetoverrideresponsebody.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.V2RatelimitSetOverrideResponseBody$ = exports.V2RatelimitSetOverrideResponseBody$outboundSchema = exports.V2RatelimitSetOverrideResponseBody$inboundSchema = void 0;
exports.v2RatelimitSetOverrideResponseBodyToJSON = v2RatelimitSetOverrideResponseBodyToJSON;
exports.v2RatelimitSetOverrideResponseBodyFromJSON = v2RatelimitSetOverrideResponseBodyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const meta_js_1 = require("./meta.js");
const ratelimitsetoverrideresponsedata_js_1 = require("./ratelimitsetoverrideresponsedata.js");
/** @internal */
exports.V2RatelimitSetOverrideResponseBody$inboundSchema = z.object({
    meta: meta_js_1.Meta$inboundSchema,
    data: ratelimitsetoverrideresponsedata_js_1.RatelimitSetOverrideResponseData$inboundSchema,
});
/** @internal */
exports.V2RatelimitSetOverrideResponseBody$outboundSchema = z.object({
    meta: meta_js_1.Meta$outboundSchema,
    data: ratelimitsetoverrideresponsedata_js_1.RatelimitSetOverrideResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var V2RatelimitSetOverrideResponseBody$;
(function (V2RatelimitSetOverrideResponseBody$) {
    /** @deprecated use `V2RatelimitSetOverrideResponseBody$inboundSchema` instead. */
    V2RatelimitSetOverrideResponseBody$.inboundSchema = exports.V2RatelimitSetOverrideResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitSetOverrideResponseBody$outboundSchema` instead. */
    V2RatelimitSetOverrideResponseBody$.outboundSchema = exports.V2RatelimitSetOverrideResponseBody$outboundSchema;
})(V2RatelimitSetOverrideResponseBody$ || (exports.V2RatelimitSetOverrideResponseBody$ = V2RatelimitSetOverrideResponseBody$ = {}));
function v2RatelimitSetOverrideResponseBodyToJSON(v2RatelimitSetOverrideResponseBody) {
    return JSON.stringify(exports.V2RatelimitSetOverrideResponseBody$outboundSchema.parse(v2RatelimitSetOverrideResponseBody));
}
function v2RatelimitSetOverrideResponseBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.V2RatelimitSetOverrideResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitSetOverrideResponseBody' from JSON`);
}
