// @unkey/api/dist/esm/sdk/ratelimit.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { ratelimitDeleteOverride } from "../funcs/ratelimitDeleteOverride.js";
import { ratelimitGetOverride } from "../funcs/ratelimitGetOverride.js";
import { ratelimitLimit } from "../funcs/ratelimitLimit.js";
import { ratelimitListOverrides } from "../funcs/ratelimitListOverrides.js";
import { ratelimitSetOverride } from "../funcs/ratelimitSetOverride.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class Ratelimit extends ClientSDK {
    async limit(request, options) {
        return unwrapAsync(ratelimitLimit(this, request, options));
    }
    async setOverride(request, options) {
        return unwrapAsync(ratelimitSetOverride(this, request, options));
    }
    async getOverride(request, options) {
        return unwrapAsync(ratelimitGetOverride(this, request, options));
    }
    async listOverrides(request, options) {
        return unwrapAsync(ratelimitListOverrides(this, request, options));
    }
    async deleteOverride(request, options) {
        return unwrapAsync(ratelimitDeleteOverride(this, request, options));
    }
}
