// @unkey/api/dist/esm/models/errors/badrequesterrorresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import * as components from "../components/index.js";
export class BadRequestErrorResponse extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.meta = err.meta;
        this.error = err.error;
        this.name = "BadRequestErrorResponse";
    }
}
/** @internal */
export const BadRequestErrorResponse$inboundSchema = z.object({
    meta: components.Meta$inboundSchema,
    error: components.BadRequestErrorDetails$inboundSchema,
})
    .transform((v) => {
    return new BadRequestErrorResponse(v);
});
/** @internal */
export const BadRequestErrorResponse$outboundSchema = z.instanceof(BadRequestErrorResponse)
    .transform(v => v.data$)
    .pipe(z.object({
    meta: components.Meta$outboundSchema,
    error: components.BadRequestErrorDetails$outboundSchema,
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var BadRequestErrorResponse$;
(function (BadRequestErrorResponse$) {
    /** @deprecated use `BadRequestErrorResponse$inboundSchema` instead. */
    BadRequestErrorResponse$.inboundSchema = BadRequestErrorResponse$inboundSchema;
    /** @deprecated use `BadRequestErrorResponse$outboundSchema` instead. */
    BadRequestErrorResponse$.outboundSchema = BadRequestErrorResponse$outboundSchema;
})(BadRequestErrorResponse$ || (BadRequestErrorResponse$ = {}));
