// @unkey/api/dist/esm/models/errors/conflicterrorresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import * as components from "../components/index.js";
export class ConflictErrorResponse extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.meta = err.meta;
        this.error = err.error;
        this.name = "ConflictErrorResponse";
    }
}
/** @internal */
export const ConflictErrorResponse$inboundSchema = z.object({
    meta: components.Meta$inboundSchema,
    error: components.BaseError$inboundSchema,
})
    .transform((v) => {
    return new ConflictErrorResponse(v);
});
/** @internal */
export const ConflictErrorResponse$outboundSchema = z.instanceof(ConflictErrorResponse)
    .transform(v => v.data$)
    .pipe(z.object({
    meta: components.Meta$outboundSchema,
    error: components.BaseError$outboundSchema,
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ConflictErrorResponse$;
(function (ConflictErrorResponse$) {
    /** @deprecated use `ConflictErrorResponse$inboundSchema` instead. */
    ConflictErrorResponse$.inboundSchema = ConflictErrorResponse$inboundSchema;
    /** @deprecated use `ConflictErrorResponse$outboundSchema` instead. */
    ConflictErrorResponse$.outboundSchema = ConflictErrorResponse$outboundSchema;
})(ConflictErrorResponse$ || (ConflictErrorResponse$ = {}));
