// @unkey/api/dist/esm/models/errors/internalservererrorresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import * as components from "../components/index.js";
export class InternalServerErrorResponse extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.meta = err.meta;
        this.error = err.error;
        this.name = "InternalServerErrorResponse";
    }
}
/** @internal */
export const InternalServerErrorResponse$inboundSchema = z.object({
    meta: components.Meta$inboundSchema,
    error: components.BaseError$inboundSchema,
})
    .transform((v) => {
    return new InternalServerErrorResponse(v);
});
/** @internal */
export const InternalServerErrorResponse$outboundSchema = z.instanceof(InternalServerErrorResponse)
    .transform(v => v.data$)
    .pipe(z.object({
    meta: components.Meta$outboundSchema,
    error: components.BaseError$outboundSchema,
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var InternalServerErrorResponse$;
(function (InternalServerErrorResponse$) {
    /** @deprecated use `InternalServerErrorResponse$inboundSchema` instead. */
    InternalServerErrorResponse$.inboundSchema = InternalServerErrorResponse$inboundSchema;
    /** @deprecated use `InternalServerErrorResponse$outboundSchema` instead. */
    InternalServerErrorResponse$.outboundSchema = InternalServerErrorResponse$outboundSchema;
})(InternalServerErrorResponse$ || (InternalServerErrorResponse$ = {}));
