// @unkey/api/dist/esm/models/errors/preconditionfailederrorresponse.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import * as components from "../components/index.js";
export class PreconditionFailedErrorResponse extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.meta = err.meta;
        this.error = err.error;
        this.name = "PreconditionFailedErrorResponse";
    }
}
/** @internal */
export const PreconditionFailedErrorResponse$inboundSchema = z.object({
    meta: components.Meta$inboundSchema,
    error: components.BaseError$inboundSchema,
})
    .transform((v) => {
    return new PreconditionFailedErrorResponse(v);
});
/** @internal */
export const PreconditionFailedErrorResponse$outboundSchema = z.instanceof(PreconditionFailedErrorResponse)
    .transform(v => v.data$)
    .pipe(z.object({
    meta: components.Meta$outboundSchema,
    error: components.BaseError$outboundSchema,
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PreconditionFailedErrorResponse$;
(function (PreconditionFailedErrorResponse$) {
    /** @deprecated use `PreconditionFailedErrorResponse$inboundSchema` instead. */
    PreconditionFailedErrorResponse$.inboundSchema = PreconditionFailedErrorResponse$inboundSchema;
    /** @deprecated use `PreconditionFailedErrorResponse$outboundSchema` instead. */
    PreconditionFailedErrorResponse$.outboundSchema = PreconditionFailedErrorResponse$outboundSchema;
})(PreconditionFailedErrorResponse$ || (PreconditionFailedErrorResponse$ = {}));
