// @unkey/api/dist/esm/models/components/apiscreateapiresponsedata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const ApisCreateApiResponseData$inboundSchema = z.object({
    apiId: z.string(),
    name: z.string(),
});
/** @internal */
export const ApisCreateApiResponseData$outboundSchema = z.object({
    apiId: z.string(),
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var ApisCreateApiResponseData$;
(function (ApisCreateApiResponseData$) {
    /** @deprecated use `ApisCreateApiResponseData$inboundSchema` instead. */
    ApisCreateApiResponseData$.inboundSchema = ApisCreateApiResponseData$inboundSchema;
    /** @deprecated use `ApisCreateApiResponseData$outboundSchema` instead. */
    ApisCreateApiResponseData$.outboundSchema = ApisCreateApiResponseData$outboundSchema;
})(ApisCreateApiResponseData$ || (ApisCreateApiResponseData$ = {}));
export function apisCreateApiResponseDataToJSON(apisCreateApiResponseData) {
    return JSON.stringify(ApisCreateApiResponseData$outboundSchema.parse(apisCreateApiResponseData));
}
export function apisCreateApiResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApisCreateApiResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApisCreateApiResponseData' from JSON`);
}
