// @unkey/api/dist/esm/models/components/livenessresponsedata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const LivenessResponseData$inboundSchema = z.object({
    message: z.string(),
});
/** @internal */
export const LivenessResponseData$outboundSchema = z.object({
    message: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var LivenessResponseData$;
(function (LivenessResponseData$) {
    /** @deprecated use `LivenessResponseData$inboundSchema` instead. */
    LivenessResponseData$.inboundSchema = LivenessResponseData$inboundSchema;
    /** @deprecated use `LivenessResponseData$outboundSchema` instead. */
    LivenessResponseData$.outboundSchema = LivenessResponseData$outboundSchema;
})(LivenessResponseData$ || (LivenessResponseData$ = {}));
export function livenessResponseDataToJSON(livenessResponseData) {
    return JSON.stringify(LivenessResponseData$outboundSchema.parse(livenessResponseData));
}
export function livenessResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => LivenessResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LivenessResponseData' from JSON`);
}
