// @unkey/api/dist/esm/models/components/ratelimitsetoverrideresponsedata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const RatelimitSetOverrideResponseData$inboundSchema = z.object({
    overrideId: z.string(),
});
/** @internal */
export const RatelimitSetOverrideResponseData$outboundSchema = z.object({
    overrideId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var RatelimitSetOverrideResponseData$;
(function (RatelimitSetOverrideResponseData$) {
    /** @deprecated use `RatelimitSetOverrideResponseData$inboundSchema` instead. */
    RatelimitSetOverrideResponseData$.inboundSchema = RatelimitSetOverrideResponseData$inboundSchema;
    /** @deprecated use `RatelimitSetOverrideResponseData$outboundSchema` instead. */
    RatelimitSetOverrideResponseData$.outboundSchema = RatelimitSetOverrideResponseData$outboundSchema;
})(RatelimitSetOverrideResponseData$ || (RatelimitSetOverrideResponseData$ = {}));
export function ratelimitSetOverrideResponseDataToJSON(ratelimitSetOverrideResponseData) {
    return JSON.stringify(RatelimitSetOverrideResponseData$outboundSchema.parse(ratelimitSetOverrideResponseData));
}
export function ratelimitSetOverrideResponseDataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RatelimitSetOverrideResponseData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RatelimitSetOverrideResponseData' from JSON`);
}
