// @unkey/api/dist/esm/models/components/v2apiscreateapiresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { ApisCreateApiResponseData$inboundSchema, ApisCreateApiResponseData$outboundSchema, } from "./apiscreateapiresponsedata.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
/** @internal */
export const V2ApisCreateApiResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: ApisCreateApiResponseData$inboundSchema,
});
/** @internal */
export const V2ApisCreateApiResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: ApisCreateApiResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2ApisCreateApiResponseBody$;
(function (V2ApisCreateApiResponseBody$) {
    /** @deprecated use `V2ApisCreateApiResponseBody$inboundSchema` instead. */
    V2ApisCreateApiResponseBody$.inboundSchema = V2ApisCreateApiResponseBody$inboundSchema;
    /** @deprecated use `V2ApisCreateApiResponseBody$outboundSchema` instead. */
    V2ApisCreateApiResponseBody$.outboundSchema = V2ApisCreateApiResponseBody$outboundSchema;
})(V2ApisCreateApiResponseBody$ || (V2ApisCreateApiResponseBody$ = {}));
export function v2ApisCreateApiResponseBodyToJSON(v2ApisCreateApiResponseBody) {
    return JSON.stringify(V2ApisCreateApiResponseBody$outboundSchema.parse(v2ApisCreateApiResponseBody));
}
export function v2ApisCreateApiResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2ApisCreateApiResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2ApisCreateApiResponseBody' from JSON`);
}
