// @unkey/api/dist/esm/models/components/v2apiscreateapirequestbody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const V2ApisCreateApiRequestBody$inboundSchema = z.object({
    name: z.string(),
});
/** @internal */
export const V2ApisCreateApiRequestBody$outboundSchema = z.object({
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2ApisCreateApiRequestBody$;
(function (V2ApisCreateApiRequestBody$) {
    /** @deprecated use `V2ApisCreateApiRequestBody$inboundSchema` instead. */
    V2ApisCreateApiRequestBody$.inboundSchema = V2ApisCreateApiRequestBody$inboundSchema;
    /** @deprecated use `V2ApisCreateApiRequestBody$outboundSchema` instead. */
    V2ApisCreateApiRequestBody$.outboundSchema = V2ApisCreateApiRequestBody$outboundSchema;
})(V2ApisCreateApiRequestBody$ || (V2ApisCreateApiRequestBody$ = {}));
export function v2ApisCreateApiRequestBodyToJSON(v2ApisCreateApiRequestBody) {
    return JSON.stringify(V2ApisCreateApiRequestBody$outboundSchema.parse(v2ApisCreateApiRequestBody));
}
export function v2ApisCreateApiRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2ApisCreateApiRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2ApisCreateApiRequestBody' from JSON`);
}
