// @unkey/api/dist/esm/models/components/v2livenessresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { LivenessResponseData$inboundSchema, LivenessResponseData$outboundSchema, } from "./livenessresponsedata.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
/** @internal */
export const V2LivenessResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: LivenessResponseData$inboundSchema,
});
/** @internal */
export const V2LivenessResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: LivenessResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2LivenessResponseBody$;
(function (V2LivenessResponseBody$) {
    /** @deprecated use `V2LivenessResponseBody$inboundSchema` instead. */
    V2LivenessResponseBody$.inboundSchema = V2LivenessResponseBody$inboundSchema;
    /** @deprecated use `V2LivenessResponseBody$outboundSchema` instead. */
    V2LivenessResponseBody$.outboundSchema = V2LivenessResponseBody$outboundSchema;
})(V2LivenessResponseBody$ || (V2LivenessResponseBody$ = {}));
export function v2LivenessResponseBodyToJSON(v2LivenessResponseBody) {
    return JSON.stringify(V2LivenessResponseBody$outboundSchema.parse(v2LivenessResponseBody));
}
export function v2LivenessResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2LivenessResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2LivenessResponseBody' from JSON`);
}
