// @unkey/api/dist/esm/models/components/v2ratelimitsetoverrideresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
import { RatelimitSetOverrideResponseData$inboundSchema, RatelimitSetOverrideResponseData$outboundSchema, } from "./ratelimitsetoverrideresponsedata.js";
/** @internal */
export const V2RatelimitSetOverrideResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: RatelimitSetOverrideResponseData$inboundSchema,
});
/** @internal */
export const V2RatelimitSetOverrideResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: RatelimitSetOverrideResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitSetOverrideResponseBody$;
(function (V2RatelimitSetOverrideResponseBody$) {
    /** @deprecated use `V2RatelimitSetOverrideResponseBody$inboundSchema` instead. */
    V2RatelimitSetOverrideResponseBody$.inboundSchema = V2RatelimitSetOverrideResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitSetOverrideResponseBody$outboundSchema` instead. */
    V2RatelimitSetOverrideResponseBody$.outboundSchema = V2RatelimitSetOverrideResponseBody$outboundSchema;
})(V2RatelimitSetOverrideResponseBody$ || (V2RatelimitSetOverrideResponseBody$ = {}));
export function v2RatelimitSetOverrideResponseBodyToJSON(v2RatelimitSetOverrideResponseBody) {
    return JSON.stringify(V2RatelimitSetOverrideResponseBody$outboundSchema.parse(v2RatelimitSetOverrideResponseBody));
}
export function v2RatelimitSetOverrideResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitSetOverrideResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitSetOverrideResponseBody' from JSON`);
}
