// @unkey/api/dist/esm/models/components/v2ratelimitgetoverrideresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
import { RatelimitOverride$inboundSchema, RatelimitOverride$outboundSchema, } from "./ratelimitoverride.js";
/** @internal */
export const V2RatelimitGetOverrideResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: RatelimitOverride$inboundSchema,
});
/** @internal */
export const V2RatelimitGetOverrideResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: RatelimitOverride$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitGetOverrideResponseBody$;
(function (V2RatelimitGetOverrideResponseBody$) {
    /** @deprecated use `V2RatelimitGetOverrideResponseBody$inboundSchema` instead. */
    V2RatelimitGetOverrideResponseBody$.inboundSchema = V2RatelimitGetOverrideResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitGetOverrideResponseBody$outboundSchema` instead. */
    V2RatelimitGetOverrideResponseBody$.outboundSchema = V2RatelimitGetOverrideResponseBody$outboundSchema;
})(V2RatelimitGetOverrideResponseBody$ || (V2RatelimitGetOverrideResponseBody$ = {}));
export function v2RatelimitGetOverrideResponseBodyToJSON(v2RatelimitGetOverrideResponseBody) {
    return JSON.stringify(V2RatelimitGetOverrideResponseBody$outboundSchema.parse(v2RatelimitGetOverrideResponseBody));
}
export function v2RatelimitGetOverrideResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitGetOverrideResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitGetOverrideResponseBody' from JSON`);
}
