// @unkey/api/dist/esm/models/components/v2ratelimitlistoverridesrequestbody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const V2RatelimitListOverridesRequestBody$inboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    cursor: z.string().optional(),
    limit: z.number().int().default(10),
});
/** @internal */
export const V2RatelimitListOverridesRequestBody$outboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    cursor: z.string().optional(),
    limit: z.number().int().default(10),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitListOverridesRequestBody$;
(function (V2RatelimitListOverridesRequestBody$) {
    /** @deprecated use `V2RatelimitListOverridesRequestBody$inboundSchema` instead. */
    V2RatelimitListOverridesRequestBody$.inboundSchema = V2RatelimitListOverridesRequestBody$inboundSchema;
    /** @deprecated use `V2RatelimitListOverridesRequestBody$outboundSchema` instead. */
    V2RatelimitListOverridesRequestBody$.outboundSchema = V2RatelimitListOverridesRequestBody$outboundSchema;
})(V2RatelimitListOverridesRequestBody$ || (V2RatelimitListOverridesRequestBody$ = {}));
export function v2RatelimitListOverridesRequestBodyToJSON(v2RatelimitListOverridesRequestBody) {
    return JSON.stringify(V2RatelimitListOverridesRequestBody$outboundSchema.parse(v2RatelimitListOverridesRequestBody));
}
export function v2RatelimitListOverridesRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitListOverridesRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitListOverridesRequestBody' from JSON`);
}
