// @unkey/api/dist/esm/models/components/v2ratelimitlistoverridesresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
import { RatelimitOverride$inboundSchema, RatelimitOverride$outboundSchema, } from "./ratelimitoverride.js";
/** @internal */
export const V2RatelimitListOverridesResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: z.array(RatelimitOverride$inboundSchema),
    pagination: Pagination$inboundSchema.optional(),
});
/** @internal */
export const V2RatelimitListOverridesResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: z.array(RatelimitOverride$outboundSchema),
    pagination: Pagination$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitListOverridesResponseBody$;
(function (V2RatelimitListOverridesResponseBody$) {
    /** @deprecated use `V2RatelimitListOverridesResponseBody$inboundSchema` instead. */
    V2RatelimitListOverridesResponseBody$.inboundSchema = V2RatelimitListOverridesResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitListOverridesResponseBody$outboundSchema` instead. */
    V2RatelimitListOverridesResponseBody$.outboundSchema = V2RatelimitListOverridesResponseBody$outboundSchema;
})(V2RatelimitListOverridesResponseBody$ || (V2RatelimitListOverridesResponseBody$ = {}));
export function v2RatelimitListOverridesResponseBodyToJSON(v2RatelimitListOverridesResponseBody) {
    return JSON.stringify(V2RatelimitListOverridesResponseBody$outboundSchema.parse(v2RatelimitListOverridesResponseBody));
}
export function v2RatelimitListOverridesResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitListOverridesResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitListOverridesResponseBody' from JSON`);
}
