// @unkey/api/dist/esm/models/components/v2ratelimitsetoverriderequestbody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const V2RatelimitSetOverrideRequestBody$inboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    duration: z.number().int(),
    identifier: z.string(),
    limit: z.number().int(),
});
/** @internal */
export const V2RatelimitSetOverrideRequestBody$outboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    duration: z.number().int(),
    identifier: z.string(),
    limit: z.number().int(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitSetOverrideRequestBody$;
(function (V2RatelimitSetOverrideRequestBody$) {
    /** @deprecated use `V2RatelimitSetOverrideRequestBody$inboundSchema` instead. */
    V2RatelimitSetOverrideRequestBody$.inboundSchema = V2RatelimitSetOverrideRequestBody$inboundSchema;
    /** @deprecated use `V2RatelimitSetOverrideRequestBody$outboundSchema` instead. */
    V2RatelimitSetOverrideRequestBody$.outboundSchema = V2RatelimitSetOverrideRequestBody$outboundSchema;
})(V2RatelimitSetOverrideRequestBody$ || (V2RatelimitSetOverrideRequestBody$ = {}));
export function v2RatelimitSetOverrideRequestBodyToJSON(v2RatelimitSetOverrideRequestBody) {
    return JSON.stringify(V2RatelimitSetOverrideRequestBody$outboundSchema.parse(v2RatelimitSetOverrideRequestBody));
}
export function v2RatelimitSetOverrideRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitSetOverrideRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitSetOverrideRequestBody' from JSON`);
}
