// @unkey/api/dist/esm/models/components/v2ratelimitdeleteoverrideresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
import { RatelimitDeleteOverrideResponseData$inboundSchema, RatelimitDeleteOverrideResponseData$outboundSchema, } from "./ratelimitdeleteoverrideresponsedata.js";
/** @internal */
export const V2RatelimitDeleteOverrideResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: RatelimitDeleteOverrideResponseData$inboundSchema,
});
/** @internal */
export const V2RatelimitDeleteOverrideResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: RatelimitDeleteOverrideResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitDeleteOverrideResponseBody$;
(function (V2RatelimitDeleteOverrideResponseBody$) {
    /** @deprecated use `V2RatelimitDeleteOverrideResponseBody$inboundSchema` instead. */
    V2RatelimitDeleteOverrideResponseBody$.inboundSchema = V2RatelimitDeleteOverrideResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitDeleteOverrideResponseBody$outboundSchema` instead. */
    V2RatelimitDeleteOverrideResponseBody$.outboundSchema = V2RatelimitDeleteOverrideResponseBody$outboundSchema;
})(V2RatelimitDeleteOverrideResponseBody$ || (V2RatelimitDeleteOverrideResponseBody$ = {}));
export function v2RatelimitDeleteOverrideResponseBodyToJSON(v2RatelimitDeleteOverrideResponseBody) {
    return JSON.stringify(V2RatelimitDeleteOverrideResponseBody$outboundSchema.parse(v2RatelimitDeleteOverrideResponseBody));
}
export function v2RatelimitDeleteOverrideResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitDeleteOverrideResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitDeleteOverrideResponseBody' from JSON`);
}
