// @unkey/api/dist/esm/models/components/v2ratelimitdeleteoverriderequestbody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const V2RatelimitDeleteOverrideRequestBody$inboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    identifier: z.string(),
});
/** @internal */
export const V2RatelimitDeleteOverrideRequestBody$outboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    identifier: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitDeleteOverrideRequestBody$;
(function (V2RatelimitDeleteOverrideRequestBody$) {
    /** @deprecated use `V2RatelimitDeleteOverrideRequestBody$inboundSchema` instead. */
    V2RatelimitDeleteOverrideRequestBody$.inboundSchema = V2RatelimitDeleteOverrideRequestBody$inboundSchema;
    /** @deprecated use `V2RatelimitDeleteOverrideRequestBody$outboundSchema` instead. */
    V2RatelimitDeleteOverrideRequestBody$.outboundSchema = V2RatelimitDeleteOverrideRequestBody$outboundSchema;
})(V2RatelimitDeleteOverrideRequestBody$ || (V2RatelimitDeleteOverrideRequestBody$ = {}));
export function v2RatelimitDeleteOverrideRequestBodyToJSON(v2RatelimitDeleteOverrideRequestBody) {
    return JSON.stringify(V2RatelimitDeleteOverrideRequestBody$outboundSchema.parse(v2RatelimitDeleteOverrideRequestBody));
}
export function v2RatelimitDeleteOverrideRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitDeleteOverrideRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitDeleteOverrideRequestBody' from JSON`);
}
