// @unkey/api/dist/esm/models/components/v2ratelimitgetoverriderequestbody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const V2RatelimitGetOverrideRequestBody$inboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    identifier: z.string(),
});
/** @internal */
export const V2RatelimitGetOverrideRequestBody$outboundSchema = z.object({
    namespaceId: z.string().optional(),
    namespaceName: z.string().optional(),
    identifier: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitGetOverrideRequestBody$;
(function (V2RatelimitGetOverrideRequestBody$) {
    /** @deprecated use `V2RatelimitGetOverrideRequestBody$inboundSchema` instead. */
    V2RatelimitGetOverrideRequestBody$.inboundSchema = V2RatelimitGetOverrideRequestBody$inboundSchema;
    /** @deprecated use `V2RatelimitGetOverrideRequestBody$outboundSchema` instead. */
    V2RatelimitGetOverrideRequestBody$.outboundSchema = V2RatelimitGetOverrideRequestBody$outboundSchema;
})(V2RatelimitGetOverrideRequestBody$ || (V2RatelimitGetOverrideRequestBody$ = {}));
export function v2RatelimitGetOverrideRequestBodyToJSON(v2RatelimitGetOverrideRequestBody) {
    return JSON.stringify(V2RatelimitGetOverrideRequestBody$outboundSchema.parse(v2RatelimitGetOverrideRequestBody));
}
export function v2RatelimitGetOverrideRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitGetOverrideRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitGetOverrideRequestBody' from JSON`);
}
