// @unkey/api/dist/esm/models/components/v2ratelimitlimitresponsebody.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { Meta$inboundSchema, Meta$outboundSchema, } from "./meta.js";
import { RatelimitLimitResponseData$inboundSchema, RatelimitLimitResponseData$outboundSchema, } from "./ratelimitlimitresponsedata.js";
/** @internal */
export const V2RatelimitLimitResponseBody$inboundSchema = z.object({
    meta: Meta$inboundSchema,
    data: RatelimitLimitResponseData$inboundSchema,
});
/** @internal */
export const V2RatelimitLimitResponseBody$outboundSchema = z.object({
    meta: Meta$outboundSchema,
    data: RatelimitLimitResponseData$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var V2RatelimitLimitResponseBody$;
(function (V2RatelimitLimitResponseBody$) {
    /** @deprecated use `V2RatelimitLimitResponseBody$inboundSchema` instead. */
    V2RatelimitLimitResponseBody$.inboundSchema = V2RatelimitLimitResponseBody$inboundSchema;
    /** @deprecated use `V2RatelimitLimitResponseBody$outboundSchema` instead. */
    V2RatelimitLimitResponseBody$.outboundSchema = V2RatelimitLimitResponseBody$outboundSchema;
})(V2RatelimitLimitResponseBody$ || (V2RatelimitLimitResponseBody$ = {}));
export function v2RatelimitLimitResponseBodyToJSON(v2RatelimitLimitResponseBody) {
    return JSON.stringify(V2RatelimitLimitResponseBody$outboundSchema.parse(v2RatelimitLimitResponseBody));
}
export function v2RatelimitLimitResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => V2RatelimitLimitResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'V2RatelimitLimitResponseBody' from JSON`);
}
