// @unkey/api/dist/commonjs/models/errors/internalservererrorresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternalServerErrorResponse$ = exports.InternalServerErrorResponse$outboundSchema = exports.InternalServerErrorResponse$inboundSchema = exports.InternalServerErrorResponse = void 0;
const z = __importStar(require("zod"));
const components = __importStar(require("../components/index.js"));
class InternalServerErrorResponse extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.meta = err.meta;
        this.error = err.error;
        this.name = "InternalServerErrorResponse";
    }
}
exports.InternalServerErrorResponse = InternalServerErrorResponse;
/** @internal */
exports.InternalServerErrorResponse$inboundSchema = z.object({
    meta: components.Meta$inboundSchema,
    error: components.BaseError$inboundSchema,
})
    .transform((v) => {
    return new InternalServerErrorResponse(v);
});
/** @internal */
exports.InternalServerErrorResponse$outboundSchema = z.instanceof(InternalServerErrorResponse)
    .transform(v => v.data$)
    .pipe(z.object({
    meta: components.Meta$outboundSchema,
    error: components.BaseError$outboundSchema,
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InternalServerErrorResponse$;
(function (InternalServerErrorResponse$) {
    /** @deprecated use `InternalServerErrorResponse$inboundSchema` instead. */
    InternalServerErrorResponse$.inboundSchema = exports.InternalServerErrorResponse$inboundSchema;
    /** @deprecated use `InternalServerErrorResponse$outboundSchema` instead. */
    InternalServerErrorResponse$.outboundSchema = exports.InternalServerErrorResponse$outboundSchema;
})(InternalServerErrorResponse$ || (exports.InternalServerErrorResponse$ = InternalServerErrorResponse$ = {}));
